[terra-avatar] Changed Shared user variant to Generic Avatar variant (#2620)
cerner/terra-core: packages/terra-avatar/docs/upgrade-guide.md

# terra-avatar Upgrade Guide
## Changes from version 2 to version 3

> 

Reviewer: @ryanthemanuel, do we want to add a 'Reason for change' section? If so, what should it say?

Reviewer 2: +1.  We should add that in the upgrade guide as well as the change log.

### Removed
* `Shared User` Variant From Avatar Variants

### Added

> * `Generic` Variant with props same as `Shared User` and with `Variant` prop to support sub-variants like `single-user`, `shared-user` and `provider`.

Reviewer 3: ```suggestion
* the `generic` subcomponent that replaces the`sharedUser` subcomponent with a new `variant` prop that can be `single-user`, `shared-user`, or `provider`.
```

Author: Thank you @nramamurth. yes that sounds more appropriate. updated here : 5f171357be04821b6b25ee4000e814fe3f5bfaca

> * the `generic` subcomponent that replaces the `sharedUser` subcomponent with a new `variant` prop that can be `single-user`, `shared-user`, or `provider`.

Reviewer 4: ```suggestion
* `generic` subcomponent  replaces the `sharedUser` subcomponent
* `generic` subcomponent contains a `variant` prop that toggles between `single-user`, `shared-user`, and `provider`.
```

Author: Thank you for the corrections..! updated [changle-log ](https://github.com/cerner/terra-core/blob/71741fd4912d5c77ebcf9d14de14ee575eb84690/packages/terra-avatar/CHANGELOG.md) like mentioned above

> * `variant` prop. this will take values for sub-variants `single-user`, `shared-user` and `provider`.

Reviewer 3: This is confusing to me as I understand that you are probably talking about the `variant` prop of `Generic` sub-component but at first glance, I felt this is in contradiction to [what was mentioned earlier](https://github.com/cerner/terra-core/blob/master/packages/terra-avatar/docs/upgrade-guide.md#removed). 

I'd remove this line altogether.

Author: Replaced line with suggested line below..!

> ### Changes to CSS Custom Properties

Reviewer: Do not include changes to css custom properties in the upgrade guide. This is an implementation detail now.

Author: should we add this to README.md..?
